fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -341,37 +324,37 @@ private CompletableFuture<Integer> removeFromAndSavePartitionMetadata(@Nonnull f
      * create a partition metadata key.
      *
      * @param groupKey group key
-     * @param timestamp timestamp
+     * @param partitionKey partitioning key
      * @return partition metadata key
      */
     @Nonnull
-    byte[] partitionMetadataKeyFromTimestamp(@Nonnull Tuple groupKey, long timestamp) {
-        return state.indexSubspace.pack(partitionMetadataKeyTuple(groupKey, timestamp));
+    byte[] partitionMetadataKeyFromPartitioningValue(@Nonnull Tuple groupKey, @Nonnull Tuple partitionKey) {
+        return state.indexSubspace.pack(partitionMetadataKeyTuple(groupKey, partitionKey));
     }
 
-    private static Tuple partitionMetadataKeyTuple(final @Nonnull Tuple groupKey, final long timestamp) {
-        // If/when we support generic tuples for the `from` we should use `addAll` here to be consistent with the data
-        // layout used here
-        return groupKey.add(PARTITION_META_SUBSPACE).add(timestamp);
+    private static Tuple partitionMetadataKeyTuple(final @Nonnull Tuple groupKey, @Nonnull Tuple partitionKey) {
+        return groupKey.add(PARTITION_META_SUBSPACE).addAll(partitionKey);
     }
 
     /**
      * save partition metadata persistently.
      *
-     * @param groupKey group key
+     * @param partitioningKey partitioning key
      * @param builder builder instance
      */
-    void savePartitionMetadata(@Nonnull Tuple groupKey, @Nonnull final LucenePartitionInfoProto.LucenePartitionInfo.Builder builder) {
+    void savePartitionMetadata(@Nonnull Tuple groupingKey,
+                               @Nonnull Tuple partitioningKey,
+                               @Nonnull final LucenePartitionInfoProto.LucenePartitionInfo.Builder builder) {
         LucenePartitionInfoProto.LucenePartitionInfo updatedPartition = builder.build();
         state.context.ensureActive().set(
-                partitionMetadataKeyFromTimestamp(groupKey, getFrom(builder)),
+                partitionMetadataKeyFromPartitioningValue(groupingKey, partitioningKey),
```

Comment:
Why is this method now taking a `partitioningKey`, rather than getting it from the `builder`?